src/credentials/CHIPCert.h

```diff
@@ -0,0 +1,460 @@
+ */
+enum
+{
+    kCertType_NotSpecified    = 0x00, /**< The certificate's type has not been specified. */
+    kCertType_General         = 0x01, /**< The certificate is of a general or non-specific type. */
+    kCertType_Node            = 0x02, /**< A CHIP node certificate. */
+    kCertType_FirmwareSigning = 0x03, /**< A CHIP firmware signing certificate. */
+    kCertType_AccessToken     = 0x04, /**< A CHIP access token certificate. */
+    kCertType_CA              = 0x05, /**< A CHIP CA certificate. */
+    kCertType_AppDefinedBase  = 0x7F, /**< Application-specific certificate types should have values >= this value. */
+};
+
+/** X.509 Certificate Key Purpose Flags
+ *
+ *    The flags order must match the enumeration order of corresponding kOID_KeyPurpose values.
+ */
+enum
+{
+    kKeyPurposeFlag_ServerAuth      = 0x01,
+    kKeyPurposeFlag_ClientAuth      = 0x02,
+    kKeyPurposeFlag_CodeSigning     = 0x04,
+    kKeyPurposeFlag_EmailProtection = 0x08,
+    kKeyPurposeFlag_TimeStamping    = 0x10,
+    kKeyPurposeFlag_OCSPSigning     = 0x20,
+};
+
+/** X.509 Certificate Key Usage Flags
+ */
+enum
+{
+    kKeyUsageFlag_DigitalSignature = 0x0001,
+    kKeyUsageFlag_NonRepudiation   = 0x0002,
+    kKeyUsageFlag_KeyEncipherment  = 0x0004,
+    kKeyUsageFlag_DataEncipherment = 0x0008,
+    kKeyUsageFlag_KeyAgreement     = 0x0010,
+    kKeyUsageFlag_KeyCertSign      = 0x0020,
+    kKeyUsageFlag_CRLSign          = 0x0040,
+    kKeyUsageFlag_EncipherOnly     = 0x0080,
+    kKeyUsageFlag_DecipherOnly     = 0x0100,
+};
+
+/** CHIP Certificate Flags
+ *
+ * Contains information about a certificate that has been loaded into a ChipCertSet object.
+ */
+enum
+{
+    kCertFlag_ExtPresent_AuthKeyId        = 0x0001,
+    kCertFlag_ExtPresent_SubjectKeyId     = 0x0002,
+    kCertFlag_ExtPresent_KeyUsage         = 0x0004,
+    kCertFlag_ExtPresent_BasicConstraints = 0x0008,
+    kCertFlag_ExtPresent_ExtendedKeyUsage = 0x0010,
+    kCertFlag_AuthKeyIdPresent            = 0x0020,
+    kCertFlag_PathLenConstPresent         = 0x0040,
+    kCertFlag_IsCA                        = 0x0080,
+    kCertFlag_IsTrusted                   = 0x0100,
+    kCertFlag_TBSHashPresent              = 0x0200,
+    kCertFlag_UnsupportedSubjectDN        = 0x0400,
+    kCertFlag_UnsupportedIssuerDN         = 0x0800,
+};
+
+/** CHIP Certificate Decode Flags
+ *
+ * Contains information specifying how a certificate should be decoded.
+ */
+enum
+{
+    kDecodeFlag_GenerateTBSHash = 0x0001,
+    kDecodeFlag_IsTrusted       = 0x0002
+};
+
+/** CHIP Certificate Validate Flags
+ *
+ * Contains information specifying how a certificate should be validated.
+ */
+enum
+{
+    kValidateFlag_IgnoreNotBefore = 0x0001,
+    kValidateFlag_IgnoreNotAfter  = 0x0002,
+    kValidateFlag_RequireSHA256   = 0x0004,
+};
+
+enum
+{
+    kNullCertTime = 0
+};
+
+/**
+ *  @class ChipDN
+ *
+ *  @brief
+ *    Represents a Distinguished Name in a CHIP certificate.
+ */
+class ChipDN
```

Comment:
Suggest: class -> struct

Since all the data members are public..

and 3 below